### PR TITLE
PDA Refactor Part 1: Phantom Blood Menace

### PR DESCRIPTION
**Pull request by ohnopigeons**
1st part of a multi-part story
Links the host PDA to the cartridge object the cartridge is in and removing the bullshit snowflake procs that were literally looping through the location to find PDA.

Newscaster Access was extra snowflakey and doesn't work in this refactor but it doesn't work in the base game either https://github.com/tgstation/tgstation/issues/28680

### CODE REVIEW

**Comment by kevinz000** [comment]
doing good work, but I find the whole idea of cartridges holding the actual functions really, really silly tbh. But then again my idea of how to solve that is a pain in the ass to code so w/e.

**Comment by ExcessiveUseOfCobblestone** [comment]
are you saying you'd have the drinking-glass switch method

**Comment by RemieRichards** [comment]
@kevinz000 the carts handling their UI is like basic oop.

File: code/game/objects/items/devices/PDA/cart.dm [hunk 1]
```diff
@@ -48,6 +49,11 @@
 	var/mob/living/simple_animal/bot/active_bot
 	var/list/botlist = list()
 
+/obj/item/weapon/cartridge/New(var/obj/item/device/pda/pda)
```
Comment (lordpidey) [inline code/game/objects/items/devices/PDA/cart.dm]: Initialize

**Review by lordpidey — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/PDA/cart.dm (hunk 1 above)
Comment (ohnopigeons) [inline code/game/objects/items/devices/PDA/cart.dm]: What's the difference?

Comment (lzimann) [inline code/game/objects/items/devices/PDA/cart.dm]: [Mainly to be easier to control when you want these things to happen(map loaded or later)](https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#prefer-initialize-over-new-for-atoms)

**Comment by kilkun** [comment]
why do you find that silly kevinz

**Comment by FantasticFwoosh** [comment]
@kevinz000 its better than distinguishing every PDA individually or having the abilities of the PDA match its color in a nonsensical way outside of the clown's PDA for progamming.

You're encouraged to go rummage for more cards in maint as loot with new functions or steal them from other people.

**Comment by octareenroon91** [comment]
Is PDA code really that starved for defines?

**Comment by kevinz000** [comment]
@FantasticFwoosh @RemieRichards it's more of "why can't PDAs ever access this without the cartridge" the reason being it's hardcoded in the cartridge, should make it datums tbh but that's just my opinion.

**Comment by RemieRichards** [comment]
No, the cartridge is the correct place.
Also "Why can't PDAs ever access this without the cartridge" uhhhhh because the software is on the cartridge? PDAs are just gameboys.

**Comment by Niknakflak** [comment]
Remie has it right

**Comment by Bawhoppen** [comment]
Niknak has it right

File: code/game/objects/items/devices/PDA/cart.dm [hunk 2]
```diff
@@ -662,17 +647,17 @@ Code:
...
-				active_bot.bot_control(command= "summon", user_turf= get_turf(usr), user_access= pda.GetAccess())
+				active_bot.bot_control(command= "summon", user_turf= get_turf(usr), user_access= host_pda.GetAccess())
 			else //Forward all other bot commands to the bot itself!
 				active_bot.bot_control(command= href_list["op"], user= usr)
 
 	if(href_list["mule"]) //MULEbots are special snowflakes, and need different args due to how they work.
 
 		active_bot.bot_control(command= href_list["mule"], user= usr, pda= 1)
 
-	generate_menu(usr)
```
Comment (octareenroon91) [inline code/game/objects/items/devices/PDA/cart.dm]: Did you make a mistake dropping this line?

Comment (ohnopigeons) [inline code/game/objects/items/devices/PDA/cart.dm]: You tell me

**Pull request merged**